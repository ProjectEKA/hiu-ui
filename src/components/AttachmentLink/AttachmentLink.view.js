import React from 'react';
import * as PropTypes from 'prop-types';
import axios from 'axios';

const AttachmentLink = ({ consentReqId, attachmentPath, linkTitle }) => {
  const openAttachment = async e => {
    e.preventDefault();

    const response = await axios({
      headers:{ Authorization: localStorage.getItem('auth-token') },
      method: 'get',
      url: `/health-information/fetch/${consentReqId}${attachmentPath}`,
      baseURL: BACKEND_BASE_URL + BACKEND_API_PATH,
      responseType: 'blob'
    });

    const blob = new Blob([response.data]);
    const fileUrl = URL.createObjectURL(blob);
    const attachmentName = attachmentPath.split('/').reverse()[0];

    const link = document.createElement('a');
    link.href = fileUrl;
    link.download = attachmentName;

    // this is necessary as link.click() does not work on the some versions of firefox
    link.dispatchEvent(
      new MouseEvent('click', { 
        bubbles: true, 
        cancelable: true, 
        view: window 
      })
    );
  };

  return <a href={'#'} onClick={openAttachment}>{linkTitle}</a>;
};

AttachmentLink.propTypes = {
  consentReqId: PropTypes.string,
  attachmentPath: PropTypes.string,
  linkTitle: PropTypes.string
};

export default AttachmentLink;
