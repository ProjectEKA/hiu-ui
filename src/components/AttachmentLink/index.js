import AttachmentLink from './AttachmentLink.view';

export default AttachmentLink;
