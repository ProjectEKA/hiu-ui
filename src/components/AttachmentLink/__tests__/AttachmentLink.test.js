import React from 'react';
import { shallow } from 'enzyme';
import AttachmentLink from '../AttachmentLink.view';

describe('AttachmentLink', () => {
  it('should render properly', () => {
    const wrapper = shallow(
      <AttachmentLink
        consentReqId="some-uuid"
        attachmentPath="/attachments/some-path"
        linkTitle="Link"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
