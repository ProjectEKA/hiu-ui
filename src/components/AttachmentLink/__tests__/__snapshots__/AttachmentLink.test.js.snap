// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AttachmentLink should render properly 1`] = `
<a
  href="#"
  onClick={[Function]}
>
  Link
</a>
`;
